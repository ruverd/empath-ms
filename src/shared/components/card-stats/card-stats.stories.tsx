import type { Meta, StoryObj } from "@storybook/react";

import { MOCK_STATS } from "@/shared/mocks/stats.mock";

import { CardStats } from "./card-stats";

const meta = {
  title: "Components/CardStats",
  component: CardStats,
  parameters: {
    layout: "centered",
  },
  decorators: [
    (Story) => (
      <div className="flex justify-center items-center gap-2 size-80 bg-shades-violet-1">
        <Story />
      </div>
    ),
  ],
  tags: ["autodocs"],
} satisfies Meta<typeof CardStats>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    ...MOCK_STATS.stats[0],
  },
};
