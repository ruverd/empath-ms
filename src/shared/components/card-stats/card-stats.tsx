import { Card, CardContent, CardHeader, CardTitle } from "../ui/card";
import type { CardStatsProps } from "./card-stats.types";

export const CardStats = ({
  icon: Icon,
  title,
  highlightText,
  auxText,
}: CardStatsProps) => {
  const renderAuxText = () => {
    if (auxText) {
      return (
        <span className="text-sm font-medium text-primary-cloud">
          {auxText}
        </span>
      );
    }

    return null;
  };

  return (
    <Card className="border border-shades-violet-2 bg-transparent px-1.5 py-2 flex flex-col-reverse">
      <CardHeader className="p-0">
        <CardTitle className="flex items-center gap-0.5 text-primary-cloud justify-center">
          <Icon />
          <span className="text-xs font-medium">{title}</span>
        </CardTitle>
      </CardHeader>
      <CardContent className="p-0 flex flex-row items-center justify-center">
        <strong className="font-bold text-lg text-shades-aqua-8">
          {highlightText}
        </strong>
        {renderAuxText()}
      </CardContent>
    </Card>
  );
};
