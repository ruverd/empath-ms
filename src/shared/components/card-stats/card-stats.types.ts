import type { Stat } from "@/shared/types/stat.types";

export interface CardStatsProps extends Omit<Stat, "status"> {}
