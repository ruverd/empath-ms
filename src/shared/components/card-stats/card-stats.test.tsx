import { render, screen } from "@testing-library/react";
import { Users } from "lucide-react";
import { describe, expect, it } from "vitest";

import { CardStats } from "./card-stats";

describe("CardStats", () => {
  it("should renders with all required props", () => {
    render(
      <CardStats icon={Users} title="Total Students" highlightText="1,234" />
    );

    expect(screen.getByText("Total Students")).toBeInTheDocument();
    expect(screen.getByText("1,234")).toBeInTheDocument();
  });

  it("should renders with auxiliary text", () => {
    render(
      <CardStats
        icon={Users}
        title="Average Rating"
        highlightText="4.8"
        auxText="/5"
      />
    );

    expect(screen.getByText("4.8")).toBeInTheDocument();
    expect(screen.getByText("/5")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <CardStats
        icon={Users}
        title="Total Students"
        highlightText="1,234"
        auxText="active"
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
