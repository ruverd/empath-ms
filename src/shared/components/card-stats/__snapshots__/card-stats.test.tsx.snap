// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardStats > should match snapshot 1`] = `
<div
  class="rounded-2xl text-card-foreground border border-shades-violet-2 bg-transparent px-1.5 py-2 flex flex-col-reverse"
>
  <div
    class="flex flex-col space-y-1.5 p-0"
  >
    <div
      class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-0.5 text-primary-cloud justify-center"
    >
      <svg
        class="lucide lucide-users"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="9"
          cy="7"
          r="4"
        />
        <path
          d="M22 21v-2a4 4 0 0 0-3-3.87"
        />
        <path
          d="M16 3.13a4 4 0 0 1 0 7.75"
        />
      </svg>
      <span
        class="text-xs font-medium"
      >
        Total Students
      </span>
    </div>
  </div>
  <div
    class="p-0 flex flex-row items-center justify-center"
  >
    <strong
      class="font-bold text-lg text-shades-aqua-8"
    >
      1,234
    </strong>
    <span
      class="text-sm font-medium text-primary-cloud"
    >
      active
    </span>
  </div>
</div>
`;
